src/terrain/draw_terrain_raster.js

```diff
@@ -171,6 +171,10 @@ function drawTerrainForGlobe(painter: Painter, terrain: Terrain, sourceCache: So
     const elevationOptions = {useDenormalizedUpVectorScale: true};
 
     batches.forEach(isWireframe => {
+        const tr = painter.transform;
+        // Use varying skirt hight applying limit for extremely large axaggerations
+        const skirtHeightValue = Math.min(skirtHeight(tr.zoom) * terrain.exaggeration(), 20000);
+
```

Comment:
How was the value "20000" chosen?

Reply:
heuristically, without the limiting factor on certain zoom levels, e.g. 1.5 +- there can be seen triangles penetrating the earth surface. It was revealed by the globe anti-aliasing test with extreme exaggerations

Reply:
on the opposite side of the globe? What are the implications of triangles penetrating the surface?

Reply:
Fixed